### PR TITLE
feat(core): update useNavigation to pass meta.query as URL params

### LINKED ISSUE #6528: [FEAT]`ListButton` from `@refinedev/mui` doesn't implement the `meta` query properly
### Describe the bug

- I tried to use the ListButton to route to the list page with some query filters but it completely overlooks the filters.
-  Here is what I did
```tsx
<ListButton
      resource="contacts"
      meta={{
        query: {
          filters: [
            {
              field: filterField,
              operator: "contains",
              value: filterValue
            }
          ]
        }
      }}
    >
      Contacts
    </ListButton>
```
- But instead of having the filters be part of the URL query, there is only `/contacts`
------ 
- I followed the component definition and the error may be how the `useNavigationButton` from `@refinedev/core` is handling the meta prop
```tsx
  const navigation = useNavigation();

  const to = React.useMemo(() => {
    if (!resource) return "";
    switch (props.action) {
      case "create":
      case "list": // here
        return navigation[`${props.action}Url`](resource, props.meta);
      default:
        if (!id) return "";
        return navigation[`${props.action}Url`](resource, id, props.meta);
    }
  }, [resource, id, props.meta, navigation[`${props.action}Url`]]);
```
- I believe the issue is in the definition of `listUrl` inside the definition of `useNavigation`, though I'm unsure since I haven't indepthly gone through the source.

### Steps To Reproduce

- Attempt to navigate to a list page using:
```tsx
<ListButton
      resource="contacts"
      meta={{
        query: {
          filters: [
            {
              field: filterField,
              operator: "contains",
              value: filterValue
            }
          ]
        }
      }}
    >
      Contacts
    </ListButton>
```

### Expected behavior

- the list button works with url query filters

### Packages

- "@refinedev/mui": "^5.14.4",
- "@refinedev/core": "^4.47.1",

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by OmkarBansod02**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://refine.dev/docs/guides-concepts/contributing/#commit-convention

## Bugs / Features

- [x] Related issue(s) linked
- [ ] Tests for the changes have been added
- [ ] Docs have been added / updated
- [x] Changesets have been added https://refine.dev/docs/guides-concepts/contributing/#creating-a-changeset

## What is the current behavior?

`listUrl` , `createUrl` , `editUrl` , `showUrl` and `cloneUrl ` functions in `useNavigation` hook doesn't properly process query filters passed through the  `meta ` parameter, resulting in missing query parameters in the generated URL

## What is the new behavior?

The functions now correctly processes and includes query parameters from the  `meta`  object when generating URLs, enabling proper query filter support for any component that uses the `useNavigation ` hook
fixes #6528

## Notes for reviewers

I’ve created a new PR for the `releases/december-2024 branch` , reapplying the changes from the previous PR [#6530 ] that was merged into main by mistake. . Please review and let me know if adjustments are needed.

### CODE REVIEW

**Review by BatuhanW — CHANGES_REQUESTED** [review]
@OmkarBansod02 thanks for re-opening the PR. Could you add github issue url to the bottom line in the changeset as it was in the previous PR?

**Comment by OmkarBansod02** [comment]
> @OmkarBansod02 thanks for re-opening the PR. Could you add github issue url to the bottom line in the changeset as it was in the previous PR?

Done 👍

File: .changeset/dry-tomatoes-beg.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+"@refinedev/core": minor
+---
+
+This PR fixes an issue where the ListButton componet doesnt include query filter in the URL navigating.
```
Comment (youneshenniwrites) [inline .changeset/dry-tomatoes-beg.md:5]:
```suggestion
This PR fixes an issue where the ListButton component doesn't include a query filter in the navigation URL.
```

**Review by BatuhanW — APPROVED** [review]
Thanks @OmkarBansod02 and @youneshenniwrites

**Pull request merged**